### PR TITLE
Block: fix invalid block scrim overflowing toolbar on mobile

### LINKED ISSUE #9402: Invalid block scrim hides block options on mobile
**Describe the bug**
If a block becomes invalid and you are viewing on mobile then the grey scrim hides the block options.

edit_post_ _wordpress_latest_ _wordpress

**To Reproduce**
1. View editor using a mobile-sized display (i.e. Chrome dev tools > Device)
2. Make a block invalid (remove the trailing `</p>` of a paragraph)
3. Scroll down to the bottom of the block
4. Note that the block dropdown menu, and other buttons, are hidden behind the grey scrim

**Expected behavior**
The dropdown menu button to be available as it is on desktop. I'm not sure about the other buttons

### PR DESCRIPTION
**Pull request by johngodley**
On small screens the invalid block scrim overflows a block’s dropdown menu / toolbar, which is positioned beneath the block.

edit_post_ _wordpress_latest_ _wordpress

This only happens when the block is selected.

This PR shortens the scrim so it doesn't overlap the bottom toolbar, as discussed in #9402.

Fixes #9402 

## Testing

1. Make a block invalid (i.e. edit HTML on a paragraph and remove the trailing `</p>`)
2. Switch browser to a mobile display
3. Ensure block is selected and verify that without this PR the scrim overflows the bottom of the block, as shown above
4. Apply PR and verify that the scrim does not cover the toolbar and dropdown menu at the bottom of the block.
edit_post_ _wordpress_latest_ _wordpress
5. Switch back to desktop  mode and verify that the scrim fully covers the block
edit_post_ _wordpress_latest_ _wordpress
5. Verify that an unselected block looks correct in both desktop and mobile mode

## Types of changes

The CSS changes are tightly coupled to a selected block on a small screen so shouldn't affect anything else.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by johngodley** [comment]
@jasmussen thanks for the help with this. I implemented your pseudo CSS but found the bottom offset wasn't enough and had to change it to

`$block-toolbar-height - $block-padding - 1px`

I don't know if there's a better constant, or why it needs the `-1px`.

I also tied the change to `.has-warning.is-selected` as the block is shorter when unselected.

**Comment by jasmussen** [comment]
Solid PR, very nice work! That one pixel is the border width, so I pushed a tiny thing to fix to just use the variable, this will help explain why it's needed.

**Review by jasmussen — APPROVED** [review]
👍 👍

**Comment by jasmussen** [comment]
Okay, so I commandeered this PR, inspired by your fixes. And I pushed a great deal of changes in https://github.com/WordPress/gutenberg/pull/9473/commits/3a6e1f3397ef1e4066965ec690b1db8b5a3917bc, enough that someone else should review this. The block warning looks like this now:

current

The big change here:

- The block warning no longer obscures content, it makes the block taller to accommodate the warning. 
- The scrim is lighter so it's easier to see what the block is.
- The hover style is not obscured by the warning

**Comment by johngodley** [comment]
I like the block growing to show all of the content.

Looks good in all warning situations, except in a single-use block warning where the border disappears in places:

edit_post_ _wordpress_latest_ _wordpress

However I think it's part of a larger issue to do with the block warning style and the single use error. I can push it out into another issue unless you think it's a quick fix here.

**Comment by johngodley** [comment]
Giving it a go I got this in `warning/style.scss`

```scss
.editor-block-list__block:not(.has-warning) .editor-warning {
	border: $border-width solid $light-gray-500;
}

.editor-block-list__block.is-selected:not(.has-warning) .editor-warning {
	border-top: $border-width solid $light-gray-500;
}
```

But it has a double border when the block is selected. Not really sure of the best way of proceeding so I'll leave it to someone who knows what they are doing.

**Comment by jasmussen** [comment]
Good catch.

I restored the full border. It's worth getting this warning in, then we can look at separate improvements, well, separately.

screen shot 2018-09-03 at 09 47 40

**Comment by mtias** [comment]
I like these changes. One thing that is jumping to me is that the option "Keep as HTML" could probably be "Keep as is" (better written, of course, maybe even just "Keep"). Being able to see the whole block clearly shows the content is not necessarily broken, just that it cannot be handled as a perfect block of a given type.

File: packages/editor/src/components/block-list/style.scss
Comment (talldan) [inline packages/editor/src/components/block-list/style.scss]: Given this is `position: relative`, I was wondering if we could just use `top:` instead of the more verbose `transform: translateY`.

**Review by talldan — APPROVED** [review]
I like this - it's a nice improvement. I added one minor optional comment.

On the topic of copy, @mtias, I find "This block has been modified externally." to be a bit of a technical sentence that could be softened and it doesn't attempt to educate why there's an issue.

And 'Convert to Block' doesn't always make sense since it's already a block, just one with an warning state.

I also noticed that 'Convert to block' appears in both the main body of the error and the more menu and I couldn't see why.

Shall I make an issue to cover those things?

**Comment by mtias** [comment]
@talldan sure, can you add to https://github.com/WordPress/gutenberg/issues/7604? That gives a general view of where we are heading.

**Comment by johngodley** [comment]
Regarding the copy in the button and message I can look at those in another PR - it probably needs a more coordinated think.

> I also noticed that 'Convert to block' appears in both the main body of the error and the more menu and I couldn't see why

Yep, I wasn't entirely sure whether this should be duplicated and just went with what was in the original mockup (although it has a different button). It's easy to remove so if it seems unnecessary I can do that elsewhere too.

File: packages/editor/src/components/block-list/style.scss
Comment (johngodley) [reply packages/editor/src/components/block-list/style.scss]: @jasmussen for thoughts

**Comment by johngodley** [comment]
> Regarding the copy in the button and message I can look at those in another PR

#9667

File: packages/editor/src/components/block-list/style.scss
Comment (jasmussen) [reply packages/editor/src/components/block-list/style.scss]: I could swear there was a reason involving margins in obscure cases, where translate doesn't affect the margins at all. The margins are essentially kept in place and the content just pulled out.

But in quick testing, it seems like `top: -15px;` accomplishes the same.

On this friday my brain has short circuited. Given I can't reproduce the issue I think I fixed by using translate, I can't find a good argument to keep it. But I have that niggling feeling that it wasn't chosen at random. What do you think?

Comment (talldan) [reply packages/editor/src/components/block-list/style.scss]: Lets keep it as it is then and get this merged.

**Comment by talldan** [comment]
@johngodley @jasmussen - I think this is good to merge, unless you can think of anything outstanding?

**Comment by jasmussen** [comment]
🎉

**Pull request merged**
